File: client/lib/models/review_model.dart
class Review {
  final String? id;
  final String userId;
  final String propertyId;
  final String type;
  final double rating;
  final String comments;

  Review({
    this.id,
    required this.userId,
    required this.propertyId,
    required this.type,
    required this.rating,
    required this.comments,
  });

  factory Review.fromJson(Map<String, dynamic> json) {
    return Review(
      id: json['_id'],
      userId: json['user_id']['_id'],
      propertyId: json['property_id'],
      type: json['type'],
      rating: (json['rating'] as num).toDouble(),
      comments: json['comments'] ?? "",
    );
  }

  Map<String, dynamic> toJson() {
    return {
      '_id': id ?? "",
      'user_id': userId,
      'property_id': propertyId,
      'type': type,
      'rating': rating.toString(),
      'comments': comments,
    };
  }
}

File: server/models/reviewModel.js
const mongoose = require('mongoose');

const reviewSchema = new mongoose.Schema({
    user_id: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User',
        required: true
    },
    property_id: {
        type: mongoose.Schema.Types.ObjectId,
        required: true
    },
    type: {
        type: String,
        enum: ['room', 'apartment', 'office'],
        required: true
    },
    rating: {
        type: Number,
        required: true,
        min: 1,
        max: 5
    },
    comments: {
        type: String,
    }
});

module.exports = mongoose.model('Review', reviewSchema);
